src/shared/idn-util.c

```diff
@@ -42,23 +42,20 @@ int dlopen_idn(void) {
         if (idn_dl)
                 return 0; /* Already loaded */
 
-        dl = dlopen("libidn.so.12", RTLD_LAZY);
-        if (!dl) {
-                /* libidn broke ABI in 1.34, but not in a way we care about (a new field got added to an
-                 * open-coded struct we do not use), hence support both versions. */
-                dl = dlopen("libidn.so.11", RTLD_LAZY);
-                if (!dl)
-                        return log_debug_errno(SYNTHETIC_ERRNO(EOPNOTSUPP),
-                                               "libidn support is not installed: %s", dlerror());
-        }
-
-        r = dlsym_many_or_warn(
-                        dl,
-                        LOG_DEBUG,
+        r = dlopen_many_sym_or_warn(&dl, "libidn.so.12", LOG_DEBUG,
                         DLSYM_ARG(idna_to_ascii_4i),
                         DLSYM_ARG(idna_to_unicode_44i),
                         DLSYM_ARG(stringprep_ucs4_to_utf8),
                         DLSYM_ARG(stringprep_utf8_to_ucs4));
+        if (r == -EOPNOTSUPP) {
+                /* libidn broke ABI in 1.34, but not in a way we care about (a new field got added to an
+                 * open-coded struct we do not use), hence support both versions. */
+                r = dlopen_many_sym_or_warn(&dl, "libidn.so.11", LOG_DEBUG,
+                                DLSYM_ARG(idna_to_ascii_4i),
+                                DLSYM_ARG(idna_to_unicode_44i),
+                                DLSYM_ARG(stringprep_ucs4_to_utf8),
+                                DLSYM_ARG(stringprep_utf8_to_ucs4));
```

Comment:
why this change? duplicating the list of functios doesn't really sound like an imprvement?

Reply:
dlopen_many_sym_or_warn() is not an interesting helper. it just combines dlopen() and dlsym_many_or_warn() into one, nothing else. The interesting bits are in dlsym_many_or_warn() and that one the old cod was already using.

Reply:
ok